import/index_java/src/org/vufind/index/UpdateDateTracker.java

```diff
@@ -149,7 +149,6 @@ protected void finalize() throws SQLException, Throwable
         insertSql.close();
         selectSql.close();
         updateSql.close();
-        super.finalize();
```

Comment:
"The finalization mechanism is inherently problematic."
https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/Object.html#finalize--

Arguably instead of just removing the call to Object.finalize() here, the entire method should be removed and reworked to close the connections in a better way.  Fixing this Java 9 issue in some sense hides a larger problem.  But that's a larger research effort IMHO and maybe is worth a Jira.

Reply:
The documentation you shared suggests using a Cleaner, which seems like it might be a viable replacement. Is it worth trying that here and replacing the entire finalize method, rather than just eliminating the call to super, which could be misleading? See:

https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/ref/Cleaner.html

If you think we should do that work separately, maybe we should just leave the super.finalize() call in place here until we have time to do that other work.